### PR TITLE
Drop Support for Ruby 2.2 and Lock RuboCop

### PR DESCRIPTION
**Pull request by jonallured**
Hi Friends!

While updating Bearden to v0.3.1, I had to [fix a test](https://github.com/artsy/bearden/pull/352/commits/0bd14ec3f6eec741b1a7a467b5c78720942b01d8) and I was confused in a couple ways, so I decided to fix those bits. I made this PR for you!! 🐻❤️🐻❤️🐻❤️

### CODE REVIEW

**Comment by dblock** [comment]
Thanks Jon, can you please fix the Rubocops too? Lock the version of rubocop and `rubocop -a ; rubocop --auto-gen-config`. Thanks!

**Comment by jonallured** [comment]
Well, I expanded the scope of this PR after @dblock's suggestion. Looks like we didn't have RuboCop locked, which meant every build on Travis could and would use a different version - I've locked this down to the latest as of today.

As a part of complying with RuboCop 0.56, the safe navigator was used, which isn't available in Ruby 2.2, so I also dropped support for that version. I feel it's time as that version of Ruby is actually EOL at this point. I think this means that merging this PR would constitute a version bump, but I didn't take the step to make that change as I assume there's a release process that would do it.

**Comment by yuki24** [comment]
@dblock @ashkan18 This is technically going to be a breaking change. We'd have to bump the major version if we wanted to follow semver very strictly.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,10 @@
-### 0.3.1 (Next)
+### (Next)
 
-* [#43](https://github.com/ashkan18/graphlient/pull/38): Allow to load and dump schema to json - [@povilasjurcys](https://github.com/povilasjurcys).
-* Your contribution here.
+* [#45](https://github.com/ashkan18/graphlient/pull/45): Drop Support for Ruby 2.2 and Lock RuboCop - [@jonallured](https://github.com/jonallured)
```
Comment (dblock) [inline CHANGELOG.md]: Danger is complaining about the missing period here at the end.

**Comment by dblock** [comment]
I think we messed up in the previous release and somehow merged broken code. The class name changed before, this is just fixing the test.

File: CHANGELOG.md (hunk 1 above)
Comment (jonallured) [reply CHANGELOG.md]: Thanks, I had no idea what it was complaining about, haha - fixed with dbdb679.

**Comment by dblock** [comment]
@jonallured If you want to challenge yourself to come up with a solution where danger-changelog can help you fix things, https://github.com/dblock/danger-changelog/issues/33

**Comment by dblock** [comment]
Seems like we're hitting https://github.com/travis-ci/travis-ci/issues/8978 on newer rubies. Try adding 

```
before_install:
  - gem update --system
```

To .travis.yml?

**Comment by jonallured** [comment]
Ha! I've been on an upgrade rampage today and have hit this a few times already but didn't know it was a tracked issue. I think 90582e1 should do it, but I've also had to do a bundle install to get the latest of that as well. We shall see!

**Comment by dblock** [comment]
Lol danger still not happy with you ;)  Put back `* Your contribution here.` into CHANGELOG

**Comment by jonallured** [comment]
I've got a good feeling about this next one!! 🤞

**Comment by dblock** [comment]
Merged

**Comment by dblock** [comment]
@jonallured Maybe take care of https://github.com/ashkan18/graphlient/issues/44?

**Pull request merged**